Source/Core/DolphinWX/RmObjAddEdit.cpp

```diff
@@ -0,0 +1,338 @@
+// Copyright 2013 Dolphin Emulator Project
+// Licensed under GPLv2
+// Refer to the license.txt file included.
+
+#include <vector>
+
+#include <wx/button.h>
+#include <wx/gbsizer.h>
+#include <wx/msgdlg.h>
+#include <wx/radiobox.h>
+#include <wx/stattext.h>
+#include <wx/textctrl.h>
+
+#include "Common/CommonTypes.h"
+#include "Core/RmObjEngine.h"
+#include "DolphinWX/RmObjAddEdit.h"
+#include "DolphinWX/WxUtils.h"
+
+CRmObjAddEdit::CRmObjAddEdit(int _selection, wxWindow* parent, wxWindowID id, const wxString& title, const wxPoint& position, const wxSize& size, long style)
+	: wxDialog(parent, id, title, position, size, style)
+{
+	selection = _selection;
+	CreateGUIControls(selection);
+
+	Bind(wxEVT_BUTTON, &CRmObjAddEdit::SaveRmObjData, this, wxID_OK);
+}
+
+CRmObjAddEdit::~CRmObjAddEdit()
+{
+}
+
+void CRmObjAddEdit::CreateGUIControls(int _selection)
+{
+	wxString currentName = _("<Insert name here>");
+
+	if (_selection == -1)
+	{
+		tempEntries.clear();
+		tempEntries.push_back(RmObjEngine::RmObjEntry(RmObjEngine::RMOBJ_16BIT, 0x00000000));
+	}
+	else
+	{
+		currentName = StrToWxStr(rmObjCodes.at(_selection).name);
```

Comment:
`.at` can be replaced with the normal array subscript accessing, assuming valid values will only ever hit this.